tests/unit/test_hyper_transformer.py

```diff
@@ -1,101 +1,676 @@
-            'object': ['foo', 'bar', None],
-            'category': [1, 2, None],
-            'bool': [True, False, None],
-            'datetime': pd.to_datetime(['1965-05-23', None, '1997-10-17']),
+            'a': [1, 2, 3],
+            'b': ['category1', 'category2', 'category3'],
+            'c': [True, False, True],
+            'd': [1.0, 2.0, 3.0]
         })
-        data['category'] = data['category'].astype('category')
-        result = hp._analyze(data)
 
-        # Asserts
-        assert isinstance(result, dict)
-        assert set(result.keys()) == {'int', 'float', 'object', 'category', 'bool', 'datetime'}
-
-        assert isinstance(result['int'], NumericalTransformer)
-        assert isinstance(result['float'], NumericalTransformer)
-        assert isinstance(result['object'], OneHotEncodingTransformer)
-        assert isinstance(result['category'], OneHotEncodingTransformer)
-        assert isinstance(result['bool'], BooleanTransformer)
-        assert isinstance(result['datetime'], DatetimeTransformer)
-
-    def test__analyze_invalid_dtype(self):
-        """Test _analyze when a list of dtypes containing an invalid dtype is passed."""
+        # Run
+        ht._update_field_types(data)
+
+        # Assert
+        expected = {'a': 'numerical', 'b': 'categorical', 'c': 'boolean', 'd': 'float'}
+        self.assertEqual(ht.field_types, expected)
+
+    @patch('rdt.hyper_transformer.load_transformer')
+    def test__fit_field_transformer(self, load_transformer_mock):
+        """Test the ``_fit_field_transformer`` method.
+
+        This tests that the ``_fit_field_transformer`` behaves as expected.
+        It should fit the transformer it is provided, loops through its
+        outputs, transform the data if the output is not ML ready and call
+        itself recursively if it can.
+
+        Setup:
+            - A mock for ``load_transformer``.
+            - A mock for the transformer returned by ``load_transformer``.
+            The ``get_output_types`` method will return two outputs, one that
+            is ML ready and one that isn't.
+
+        Input:
+            - A DataFrame with one column.
+            - A column name to fit the transformer to.
+            - A transformer.
+
+        Output:
+            - A DataFrame with columns that result from transforming the
+            outputs of the original transformer.
+            - ``_output_columns`` should add the appropriate column names.
+            - ``_temp_columns`` should add the appropriate column names.
+        """
         # Setup
-        hp = HyperTransformer(dtypes=['int', 'complex'])
+        data = pd.DataFrame({'a': [1, 2, 3]})
+        transformed_data1 = pd.DataFrame({
+            'a.out1': [2, 4, 6],
+            'a.out2': ['1', '2', '3']
+        })
+        transformer1 = Mock()
+        transformer2 = Mock()
+        transformer1.get_output_types.return_value = {
+            'a.out1': 'categorical',
```

Comment:
Is this correct? In the DF above it seems like `a.out1` is `numerical` and `out2` `categorical`

Reply:
good catch!